### PR TITLE
feat: expose MethodInjector and defer `decode/encode` to `u128` if it is defined

### PR DESCRIPTION
**Pull request by willemneal**

### CODE REVIEW

File: core/assembly/deserializer.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export abstract class Deserializer<I> {
     let value: T
 
     // @ts-ignore
-    if (value instanceof u128) { return this.decode_u128(); }  // -> we need to get ride of this
```
Comment (gagdiez) [inline core/assembly/deserializer.ts:112]: Should we add a test in which this breaks?

File: borsh/assembly/serializer.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export class BorshSerializer extends Serializer<ArrayBuffer> {
   }  */
     if (t != null) {
       this.buffer.store<u8>(1);
-      this.encode(<NonNullable<T>>t);
```
Comment (gagdiez) [inline borsh/assembly/serializer.ts:51]: Should we add a test in which this breaks?

Comment (willemneal) [reply borsh/assembly/serializer.ts:51]: If you can find one!

File: core/assembly/deserializer.ts (hunk 1 above)
Comment (willemneal) [reply core/assembly/deserializer.ts:112]: Yeah if you can think of one!

**Pull request merged**